sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/LoadBalancerOutboundRule.java

```diff
@@ -0,0 +1,239 @@
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithFrontend<ParentT> {
+            /**
+             * Specifies a frontend for outbound rule to apply to
+             * @param name the frontend name
+             * @return the next stage of the parent definition to return to after attaching this definition
+             */
+            LoadBalancerOutboundRule.DefinitionStages.WithAttach<ParentT> toFrontend(String name);
+
+            /**
+             * Specifies a list of frontends for outbound rule to apply to
+             * @param names a list of frontend names
+             * @return the next stage of the parent definition to return to after attaching this definition
+             */
+            LoadBalancerOutboundRule.DefinitionStages.WithAttach<ParentT> toFrontends(List<String> names);
+        }
+
+        /**
+         * The stage of an outbound rule definition allowing to specify the idle connection timeout for this outbound
+         * rule.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithIdleTimeout<ParentT> {
+            /**
+             * Specifies the idle connection timeout in minutes.
+             *
+             * @param minutes a number of minutes
+             * @return the next stage of the definition
+             */
+            LoadBalancerOutboundRule.DefinitionStages.WithAttach<ParentT> withIdleTimeoutInMinutes(int minutes);
+        }
+
+        /**
+         * The stage of an outbound rule definition allowing to specify the TCP reset enablement for this outbound
+         * rule.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithEnableTcpReset<ParentT> {
+            /**
+             * Specifies the TCP reset enablement for this outbound rule
+             * @param enableTcpReset
+             * @return the next stage of the definition
+             */
+            LoadBalancerOutboundRule.DefinitionStages.WithAttach<ParentT> withEnableTcpReset(boolean enableTcpReset);
+        }
+    }
+
+    /**
+     * The entirety of an outbound rule definition.
+     *
+     * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+     */
+    interface Definition<ParentT>
+        extends LoadBalancerOutboundRule.DefinitionStages.Blank<ParentT>,
+        LoadBalancerOutboundRule.DefinitionStages.WithProtocol<ParentT>,
+        LoadBalancerOutboundRule.DefinitionStages.WithBackend<ParentT>,
+        LoadBalancerOutboundRule.DefinitionStages.WithFrontend<ParentT>,
+        LoadBalancerOutboundRule.DefinitionStages.WithAttach<ParentT> {
+    }
+
+    /** Grouping of outbound rule update stages. */
+    interface UpdateStages {
```

Comment:
Would be better use `UpdateStages<ParentT>` like `Definition`. Then it could be re-used by other parent resource.

Reply:
updated